### PR TITLE
Changes to support OCI with flux in the UI

### PR DESCRIPTION
**Pull request by antgamdia**
### Description of the change

Since the OCI support in Flux was something in progress, we didn't allow adding OCI repos in the past release. As we are implementing this support, the UI should also support it.

In this PR I'm also squeezing a fix for a regression we've had: Flux repos couldn't be created as they need to be global. This was broken in the https://github.com/vmware-tanzu/kubeapps/pull/5203 hotfix.

Besides, this PR fixes several minor problems, namely:

- Rename to `setInterval` to `setSyncInterval` since there is a js function called setInterval and the IDE was triggering some warnings.
- Avoid re-auto-selecting storage type if the user already has selected one.
- Avoid overriding the scheme (`https://`, `oci://`, etc) with the default one except for the helm plugin. It was causing some issues with the Flux OCI repos.

### Benefits

See above.

### Possible drawbacks

N/A

### Applicable issues

- related #5007 

### Additional information

I haven't been able to fetch packages from an OCI repo yet IRL, but the repo CR is being properly created and used, according to the logs.

### CODE REVIEW

File: dashboard/src/actions/repos.ts [hunk 1]
```diff
@@ -139,7 +139,12 @@ export const addRepo = (
     } = getState();
     try {
       dispatch(addOrUpdateRepo());
-      const namespaceScoped = !isGlobalNamespace(namespace, request.plugin?.name, config);
+      // TODO(agamez): improve the global/namespaced repos UX
+      // to avoid hacks like this (flux repos are always global)
+      // https://github.com/vmware-tanzu/kubeapps/issues/2995
+      const namespaceScoped =
+        !isGlobalNamespace(namespace, request.plugin?.name, config) &&
+        request.plugin?.name !== PluginNames.PACKAGES_FLUX;
```
Comment (castelblanque) [inline dashboard/src/actions/repos.ts:147]: `isGlobalNamespace` has a switch per plugin.
Why not just returning `true` for the Flux case inside that function?

Comment (antgamdia) [reply dashboard/src/actions/repos.ts:147]: Yep, good point. I guess it makes sense for the `isGlobalNamespace` function to always return `true` for flux packages. I'll do it.

Comment (antgamdia) [reply dashboard/src/actions/repos.ts]: Done at https://github.com/vmware-tanzu/kubeapps/pull/5218/commits/3f8b65c8551ec387bcfdd9ee9fe7386d2813fe28

**Review by castelblanque — APPROVED** [review]
LGTM

**Pull request merged**